jina/orchestrate/pods/config/k8s.py

```diff
@@ -354,7 +352,9 @@ def to_k8s_yaml(
             .. # noqa: DAR201
             .. # noqa: DAR101
         """
-        if self.name == 'gateway':
+        if hasattr(self.args, 'external') and self.args.external:
+            return []
```

Comment:
here as well

Reply:
thats not a Pod, but a K8sPodConfig, I think it has no `external` property and it also does not make much sense to add here. An external K8sPodConfig should never be created in the first place (and it is not with the current code), this check is just to double check this behaviour.

Reply:
Ah yes, sure! then this check is not needed at all in theory right?
I would just have an `assert` at instantiation time